flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClient.java

```diff
@@ -219,6 +230,68 @@ public KubernetesWatch watchPodsAndDoCallback(
 				.watch(new KubernetesPodsWatcher(podCallbackHandler)));
 	}
 
+	@Override
+	public CompletableFuture<Void> createConfigMap(KubernetesConfigMap configMap) {
+		return CompletableFuture.runAsync(
+			() -> {
+				if (!getConfigMap(configMap.getName()).isPresent()) {
+					this.internalClient.configMaps().create(configMap.getInternalResource());
+				}
```

Comment:
If the ConfigMap already exists, the create operation will fail. Since the check-and-create is not an atomic operation, I will use a `try... catch` to directly creating the ConfigMap.